src/Libraries/CoreNodeModels/Input/BaseTypes.cs

```diff
@@ -103,12 +103,29 @@ public DoubleInput()
 
             ShouldDisplayPreviewCore = false;
             ConvertToken = Convert;
-            Value = "0";
+            NumericalValue = "0";
 
             //ws.DynamoModel.PreferenceSettings.PropertyChanged += Preferences_PropertyChanged;
         }
 
-        public virtual double Convert(double value)
+        //if the string can be parsed to a common numeric type return true
+        private bool validateInput(string value)
+        {
+            bool canConvert = false;
+
+            int intVal;
```

Comment:
I believe int is by default 32digit in C#, see https://msdn.microsoft.com/en-us/library/exx3b86w.aspx